File: src/lec08/Lec08Main.java
package lec08;

public class Lec08Main {
    public static void main(String[] args) {
        String[] strArr = {"A, B, C"};
        printAll(strArr);
        printAll("A, B, C");
    }

    public static void printAll(String... strings) {
        for (String str : strings) {
            System.out.println(str);
        }
    }

    public int max(int a, int b) {
        if (a > b) {
            return a;
        }
        return b;
    }

    public void repeat(String str, int num, boolean userNewLine) {
        for (int i = 1; i <= num; i++) {
            if (userNewLine) {
                System.out.println(str);
            } else {
                System.out.print(str);
            }
        }
    }

    // 오버로딩 활용 - userNewLine이 항상 true
    public void repeat(String str, int num) {
        repeat(str, num, true);
    }

    // 오버로딩 활용 - userNewLine이 항상 true이고 반복횟수가 3으로 고정
    public void repeat(String str) {
        repeat(str, 3, true);
    }

}

File: src/lec08/Lec08Main.kt
package lec08

fun max(a: Int, b: Int): Int {
    return if (a > b) {
        a
    } else {
        b
    }
}

fun max2(a: Int, b: Int) =
    if (a > b) {
        a
    } else {
        b
    }

fun max3(a: Int, b: Int) = if (a > b) a else b


fun main() {
    repeat("Hello World", 3, true)
    repeat("Hello World", 3)
    repeat("Hello World")


    repeat("Hello World", 3, false)
    repeat("Hello World", userNewLine = false)
}

fun repeat(
    str: String,
    num: Int = 3,
    userNewLine: Boolean = true
) {
    for (i in 1..num) {
        if (userNewLine) {
            println(str)
        } else {
            print(str)
        }
    }
}